### PR TITLE
♻️ Node ports package: remove globals for user,project and node ids

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

The node ports package prior to version 1.0.0 used to need globals for USER_ID, PROJECT_ID and NODE_ID to allow access to the node ports data.
- This PR refactors it so that these variables are passed on the node_ports instance.
- Also the node_ports filemanager now downloads files to ```/tmp/simcorefiles/THREAD_ID/key/``` instead of ```/tmp/simcorefiles/key/```

This allows the node_ports to be used in a threadpool and is useful for the dask-workers and other asyncio.gather calls.
NOTE: the version was bumped to the next major. That is ```1.0.0```. for legacy services that are not fixing their version of the simcore-sdk that might need to fix it in order to re-compile them.

Before:
```python
node_config.USER_ID = str(user_id)
node_config.NODE_UUID = str(node_uuid)
node_config.PROJECT_ID = str(project_id)
PORTS = await node_ports.ports()
```

After:
```python
PORTS = await node_ports.ports(user_id, project_id, node_uuid)
```

Bonus:
- removed some old crap (ORM models)

## How to test

```bash
make build
cd packages/simcore-sdk
make install-dev
make tests
```

### CODE REVIEW

File: packages/simcore-sdk/tests/integration/np_helpers.py [hunk 1]
```diff
@@ -8,36 +8,36 @@
...
-        ComputationalTask.node_id == str(node_uuid),
-    )
-    task.update(
-        dict(
-            schema=new_configuration["schema"],
-            inputs=new_configuration["inputs"],
-            outputs=new_configuration["outputs"],
+
+    with postgres_engine.connect() as conn:
+        conn.execute(
```
Comment (pcrespov) [inline packages/simcore-sdk/tests/integration/np_helpers.py:29]: nice upgrade! :-)

File: packages/simcore-sdk/tests/integration/np_helpers.py [hunk 2]
```diff
@@ -8,36 +8,36 @@
...
+            comp_tasks.update()  # pylint: disable=no-value-for-parameter
+            .where(
+                (comp_tasks.c.project_id == project_id)
+                & (comp_tasks.c.node_id == node_uuid)
+            )
+            .values(
+                schema=new_configuration["schema"],
```
Comment (pcrespov) [inline packages/simcore-sdk/tests/integration/np_helpers.py:36]: I still think is more clear the way you write it but for one-liners ... :-)
```python
.values( **{ c:new_configuration[c] for c in ("schema", "inputs", "outputs") } )
```

File: services/director-v2/src/simcore_service_director_v2/utils/dags.py [hunk 3]
```diff
@@ -69,12 +71,19 @@ async def compute_node_modified_state(
     nodes_data_view: nx.classes.reportviews.NodeDataView, node_id: NodeID
 ) -> bool:
     node = nodes_data_view[str(node_id)]
+    # if the node state is in the modified state already
+    if not node["state"] or node["state"] in [
```
Comment (pcrespov) [inline services/director-v2/src/simcore_service_director_v2/utils/dags.py:75]: can ``bool(node['state']) == False`` e.g. ``0, None, False, ''`` etc ??
would it be more clear as
```python
if node['state'] in (None, RunningState.ABORTED, RunningStateFAILED):
    return True
```

**Review by pcrespov — APPROVED** [review]
:+1:

File: packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py [hunk 4]
```diff
@@ -41,47 +63,53 @@ async def push(file_or_folder: Path, rename_to: Optional[str] = None):
...
+        user_id=user_id,
+        store_id="0",
+        s3_object=s3_object,
+        local_folder=file_path.parent,
     )
     if downloaded_file != file_path:
         if file_path.exists():
             file_path.unlink()
-        move(downloaded_file, file_path)
+        move(f"{downloaded_file}", file_path)
```
Comment (colinRawlings) [inline packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py:81]: just out of curiousity is there a difference between `f"{var}"` and str(var)

File: packages/simcore-sdk/src/simcore_sdk/node_ports/_item.py [hunk 5]
```diff
@@ -1,25 +1,29 @@
...
 
 log = logging.getLogger(__name__)
 
 
-ItemConcreteValue = Union[int, float, bool, str, Path]
-NodeLink = Dict[str,str]
-StoreLink = Dict[str,str]
-DownloadLink = Dict[str,str]
+ItemConcreteValue = Union[int, float, bool, str, Path, Dict[str, str]]
+NodeLink = Dict[str, str]
```
Comment (colinRawlings) [inline packages/simcore-sdk/src/simcore_sdk/node_ports/_item.py:16]: Would it be worth to alias the type `Dict[str, str]`?

Comment (colinRawlings) [reply packages/simcore-sdk/src/simcore_sdk/node_ports/_item.py:16]: Ah sorry now see this line was not changed ...

**Review by colinRawlings — APPROVED** [review]
Quite far from the details but looks ok to me ...

File: packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py (hunk 4 above)
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py:81]: you mean performance wise? I'm not sure. but I like the f strings better;)

File: packages/simcore-sdk/src/simcore_sdk/node_ports/_item.py (hunk 5 above)
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports/_item.py:16]: yes you are right. this is a pretty old piece of code. the node ports package v1 I do not want to change too much.

**Pull request merged**